site/hooks/useBalance.js

```diff
@@ -0,0 +1,35 @@
+import { useContext } from 'react'
+import { useWeb3React } from '@web3-react/core'
+import useSWR from 'swr'
+import PureContext from '../components/context/Pure'
+
+export const useBalance = function ({ symbol }) {
+  const { active, account: userAccount, library } = useWeb3React()
+  const { erc20 } = useContext(PureContext)
+
+  const { data, mutate, error } = useSWR(
+    [`${symbol}-Balance-${userAccount}`, userAccount, erc20],
+    async function (_, account, erc20Instancer) {
```

Comment:
Why do we need to mark this function as `async`?

Reply:
If I understand correctly, that would help to catch the error in the sync code of this function that is returning a Promise (for instance if there's a null pointer). Isn't that correct?

Reply:
I don't think is needed. Doesn't hurt either.